### ISSUE TITLE
Min and max can be fooled by .pointValues

### ISSUE THREAD

**Issue by nickhale** [chebfun/chebfun#1959]
The following code determines that the max of this chebfun `f` is `1` at `x = -1`. However, when we evaluate `f(-1)` we get 0 because of the `.pointValues`.

```
>> f = chebfun(@(x) x.^2);
>> f([-1, 1]) = 0;
>> [y, x] = max(f)
y =
     1
x =
    -1
>> f(x)
ans =
     0
```

**Comment by ajt60gaibb** [comment]
Yes, I think only `chebfun/feval` respects the `.pointValues` property. This causes trouble in `separableApprox` too where we ended up using much slower algorithms to avoid breaking encapsulation.
